overlord/snapstate/snapstate.go

```diff
@@ -1118,14 +1083,37 @@ func ensureInstallPreconditions(st *state.State, info *snap.Info, flags Flags, s
 	return flags, nil
 }
 
+// A PrereqTracker helps tracking snap prerequisites for one or across
+// multiple snap operations. Depending of usage context implementatiomns
+// can be stateful or stateless.
+// Functions taking a PrereqTracker accept nil and promise to call
+// Add once for any target snap.
+type PrereqTracker interface {
+	// Add adds a snap for tracking.
+	Add(*snap.Info)
+	// DefaultProviderContentAttrs returns a map keyed by the names of all
+	// default-providers for the content plugs that the given snap.Info
+	// needs. The map values are the corresponding content tags.
+	// If repo is not nil, any content tag provided by an existing slot in it
+	// is considered already available and filtered out from the result.
+	// info might or might have not been passed already to Add.
+	DefaultProviderContentAttrs(info *snap.Info, repo snap.InterfaceRepo) map[string][]string
+}
+
+func prereqTrack(prqt PrereqTracker, info *snap.Info) {
```

Comment:
nitpick: I find the name a bit confusing, maybe something like `addToPrereckTracker()` would help?